### ISSUE TITLE
Wrong join tables generated when superclass has @ManyToMany relationship

### ISSUE THREAD

**Issue by mirkoluchi** [requery/requery#461]
I have a data model containing:
* a **superclass** (SuperClass)
* **2 subclasses** of that superclass (SubClassA and SubClassB)
* the superclass has a **many-to-many relationship** towards a relatedclass (RelatedClass)

```Java
@MappedSuperclass
public interface ISuperClass {
    @Id
    public Long getId();

    @ManyToMany
    @JoinTable
    public List<IRelatedClass> getRelatedClasses();
}

@Entity
public interface ISubClassA extends ISuperClass {
}

@Entity
public interface ISubClassB extends ISuperClass{
}

@Entity
public interface IRelatedClass {
    @Id
    public Long getId();
}

```

The **classes generated** by the requery annotation processor to map the **join tables** of the many-to-many relationship are **incorrect**:

```Java
@Entity(
        model = "interfaces",
        stateless = false
)
@Table(
        name = "ISubClassB_IRelatedClass"
)
public abstract class AbstractSubClassB_RelatedClass implements Serializable {
    @ForeignKey(
            delete = ReferentialAction.CASCADE,
            update = ReferentialAction.CASCADE,
            references = SubClassA.class
    )
    @Key
    protected Integer ISubClassAId;

    @ForeignKey(
            delete = ReferentialAction.CASCADE,
            update = ReferentialAction.CASCADE,
            references = RelatedClass.class
    )
    @Key
    protected Long IRelatedClassId;
}


@Entity(
        model = "interfaces",
        stateless = false
)
@Table(
        name = "ISubClassA_IRelatedClass"
)
public abstract class AbstractSubClassA_RelatedClass implements Serializable {
    @ForeignKey(
            delete = ReferentialAction.CASCADE,
            update = ReferentialAction.CASCADE,
            references = SubClassA.class
    )
    @Key
    protected Long ISubClassAId;

    @ForeignKey(
            delete = ReferentialAction.CASCADE,
            update = ReferentialAction.CASCADE,
            references = RelatedClass.class
    )
    @Key
    protected Long IRelatedClassId;
}
```

Notice in fact that **both AbstractSubClassA_RelatedClass and AbstractSubClassB_RelatedClass** have a **reference** to the **same subclass** SubClassA, while one should have a reference to SubClassA and the other to SubClassB.

_NOTE: As a workaround I have overridden the getter of the many-to-many relationship in both SubClassA and SubClassB, which made the annotation processor behave correctly._

**Comment by npurushe** [comment]
The current implementation of superclasses doesn't support relations at the moment but it wasn't explicitly made an error. You can work around it by duplicating the relational attribute in the parent entity. Looks like it won't be a large change to fix this though.

**Comment by npurushe** [comment]
Thanks for the detailed issue, should be able to include this in the next update

**Comment by npurushe** [comment]
This fix is in 1.2.1
